pkg/smbios/table.go

```diff
@@ -36,11 +36,49 @@ const (
 	outOfSpec = "<OUT OF SPEC>"
 )
 
+// ModifyStrings modifies the strings part of the table
+func (t *Table) ModifyStrings(newStr []string) {
+	t.strings = newStr
+}
+
+// ModifyData modifies the data part of the table
+func (t *Table) ModifyData(newData []byte) {
```

Comment:
This seems pretty unsafe on its own. I'd rather give callers primitives that change the data in a safe way. IIUC the requirement is only to change some of the strings, so is this needed?

Reply:
This is needed when the `Manufacturer`, `Product Name` ..etc number is more than one.

Because this field contains byte for "number of string" e.g.,  [SMBIOS spec](https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.5.0.pdf)
<img width="711" alt="Screenshot 2024-01-29 at 6 30 21 PM" src="https://github.com/u-root/u-root/assets/14128307/76372863-ea14-4dab-a8a4-248ab0b214cb">

Reply:
"change", mean adding new strings here. If a Type 1  field is not specified, there is not a place holder in there, in the strings list section. The offdet bit is always reserved ofc.